src/coffee/sync/utils/order.coffee

```diff
@@ -117,6 +117,37 @@ class OrderUtils extends BaseUtils
       .value()
     _.flatten actions
 
+  # Private: map line items
+  #
+  # diff - {Object} The result of diff from `jsondiffpatch`
+  # old_obj - {Object} The existing order
+  #
+  # Returns {Array} The list of actions, or empty if there are none
+  actionsMapLineItems: (diff, old_obj, new_obj) ->
+    actions = []
+    
+    data = new_obj || old_obj
+    
+    # Note the lineItem.state contains 'from/to' instead of just the data
+
+    if data.lineItems
+      data.lineItems.forEach (lineItem) ->
+        if (lineItem.state)
+          lineItem.state.forEach (state) ->
+            if (state.fromState && state.toState)
```

Comment:
Since this is an object and empty object is truthy you might want to check the size or values in the object.